pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -1172,3 +1172,42 @@ def _decomposition_with_one_worker(control_wires, target_wire, work_wire):
         ]
 
         return gates
+
+
+class Barrier(Operation):
+    r"""Barrier(wires)
+    The Barrier operator, used to separate the compilation process into blocks or as a visual tool.
+
+    **Details:**
+
+    * Number of wires: AnyWires
+    * Number of parameters: 0
+
+    Args:
+        only_visual (bool): True if we do not want it to have an impact on the compilation process. Default is False.
+        wires (Sequence[int] or int): the wires the operation acts on
+    """
+    num_params = 0
+    num_wires = AnyWires
+    par_domain = None
+
+    def __init__(self, only_visual=False, wires=Wires([]), do_queue=True, id=None):
+        self.only_visual = only_visual
+        super().__init__(wires=wires, do_queue=do_queue, id=id)
+
+    def decomposition(self, wires):
+        _wires = wires
+        if self.only_visual:
+            decomp_ops = []
+        else:
+            raise NotImplementedError
```

Comment:
Why the `NotImplementedError`?

Reply:
It is the way that I have that it behaves as if I had not put a decomposition since it is the behavior of the parent class. If I return `qml.Barrier()`, which would be logical, an infinite loop is formed.

Reply:
What's wrong with `[]`?

Reply:
If I run this with:
```python
def qfunc():
    qml.Hadamard(wires=0)
    qml.Barrier(wires = range(2))
    qml.Hadamard(wires=0)
    qml.Hadamard(wires=1)
    return qml.expval(qml.PauliZ(0))


dev = qml.device('default.qubit', wires=3)

optimized_qfunc = qml.compile()(qfunc)
optimized_qnode = qml.QNode(optimized_qfunc, dev)
print(qml.draw(optimized_qnode)())
```

I should get:
0: ──H──H──┤ ⟨Z⟩ 
 1: ──H─────┤   

because Barrier stop the simplification between Hadamards. 
However, if `[]`the output is:

0: ──────┤ ⟨Z⟩ 
 1: ──H───┤